src/main/java/frc/robot/subsystems/conveyor/Conveyor.java

```diff
@@ -0,0 +1,236 @@
+package frc.robot.subsystems.conveyor;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import com.revrobotics.ColorMatch;
+import com.revrobotics.ColorMatchResult;
+import com.revrobotics.ColorSensorV3;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.I2C;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Ports;
+
+import java.util.Arrays;
+import java.util.concurrent.ArrayBlockingQueue;
+
+import static frc.robot.Ports.Conveyor.INVERSION;
+
+public class Conveyor extends SubsystemBase {
+    private static Conveyor INSTANCE = null;
+    private final WPI_TalonFX motor = new WPI_TalonFX(Ports.Conveyor.MOTOR);
+    private final ArrayBlockingQueue<String> position = new ArrayBlockingQueue<>(2);
+    private final ColorSensorV3 colorSensorIntake = new ColorSensorV3(I2C.Port.kMXP);
+    private final DigitalInput postFlapBeam = new DigitalInput(Ports.Conveyor.POST_FLAP_BEAM_BREAKER);
+    private final DigitalInput preFlapBeam = new DigitalInput(Ports.Conveyor.PRE_FLAP_BEAM_BREAKER);
+    //    private final Solenoid flap = new Solenoid(PneumaticsModuleType.CTREPCM, Ports.Conveyor.SOLENOID);
+    private final ColorMatch match = new ColorMatch();
+    private DriverStation.Alliance lastSeenColor = DriverStation.Alliance.Invalid;
+    private boolean lastPostFlapBeamInput = true;
+
+    private Conveyor() {
+        motor.setInverted(INVERSION);
+        match.addColorMatch(Constants.Conveyor.RED);
+        match.addColorMatch(Constants.Conveyor.BLUE);
+        match.addColorMatch(Constants.Conveyor.NONE);
+        position.add(DriverStation.Alliance.Invalid.name());
+        position.add(DriverStation.Alliance.Invalid.name());
+    }
+
+    /**
+     * lazy instantiation
+     *
+     * @return the subsystem instance
+     */
+    public static Conveyor getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new Conveyor();
+        }
+        return INSTANCE;
+    }
+
+    /**
+     * @return the color sensor value as a DriverStation.Alliance enum.
+     */
+    public DriverStation.Alliance getColor() {
+        Color color = colorSensorIntake.getColor();
+        ColorMatchResult result = match.matchClosestColor(color);
+        Color resultColor = result.color;
+        SmartDashboard.putBoolean("match", resultColor == Color.kRed);
+
+        if (resultColor == Constants.Conveyor.RED) {
+            return DriverStation.Alliance.Red;
+        } else if (resultColor == Constants.Conveyor.BLUE) {
+            return DriverStation.Alliance.Blue;
+        } else {
+            return DriverStation.Alliance.Invalid;

```

Comment:
Why do you need to use WPILib's class? Why's `matchClosestColor()`'s result not enough?

Reply:
You mean the `DriverStation.Alliance`?